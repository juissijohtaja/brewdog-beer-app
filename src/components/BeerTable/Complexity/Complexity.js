import React from 'react'
import Box from '@material-ui/core/Box'
import { makeStyles } from '@material-ui/core/styles'
import { orange, red, lightGreen, grey } from '@material-ui/core/colors'

const useStyles = makeStyles((theme) => ({
  tag: {
    borderRadius: 4,
    fontSize: '0.8em',
    textTransform: 'uppercase',
    fontWeight: 500,
    letterSpacing: 0.7,
    paddingTop: 1,
    paddingBottom: 1,
  },
  simple: {
    color: lightGreen[800],
    backgroundColor: lightGreen[50],
    width: 64,
  },
  moderate: {
    color: orange[800],
    backgroundColor: orange[50],
    width: 80,
  },
  hazardous: {
    color: red[800],
    backgroundColor: red[50],
    width: 90,
  },
  default: {
    color: orange[800],
    backgroundColor: orange[50],
    width: 90,
  },
}))

const Complexity = ({ complexity }) => {
  const classes = useStyles()
  switch(complexity) {
  case 'Simple':
    return <Box className={`${classes.tag} ${classes.simple}`} data-testid='complexity'>
      {complexity}
    </Box>
  case 'Moderate':
    return <Box className={`${classes.tag} ${classes.moderate}`} data-testid='complexity'>
      {complexity}
    </Box>
  case 'Hazardous':
    return <Box className={`${classes.tag} ${classes.hazardous}`} data-testid='complexity'>
      {complexity}
    </Box>
  default:
    return <Box className={`${classes.tag} ${classes.default}`} data-testid='default'>
      {complexity}
    </Box>
  }
}

export default Complexity