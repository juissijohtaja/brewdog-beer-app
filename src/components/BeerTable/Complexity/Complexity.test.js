import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import initialStoreState from '../../../setupTests'
import Complexity from './Complexity'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

const mockStore = configureStore([])

describe('Complexity Component', () => {
  let store, component

  beforeEach(() => {
    store = mockStore(initialStoreState)
    store.dispatch = jest.fn()
  })

  test('should render Simple', () => {
    component = renderer.create(
      <Provider store={store}>
        <Complexity complexity='Simple' />
      </Provider>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('should render Moderate', () => {
    component = renderer.create(
      <Provider store={store}>
        <Complexity complexity='Moderate' />
      </Provider>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('should render Hazardous', () => {
    component = renderer.create(
      <Provider store={store}>
        <Complexity complexity='Hazardous' />
      </Provider>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

})