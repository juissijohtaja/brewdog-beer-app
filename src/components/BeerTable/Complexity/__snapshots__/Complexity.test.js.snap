// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Complexity Component should render Hazardous 1`] = `
<div
  className="MuiBox-root MuiBox-root-7 makeStyles-tag-1 makeStyles-hazardous-4"
  data-testid="complexity"
>
  Hazardous
</div>
`;

exports[`Complexity Component should render Moderate 1`] = `
<div
  className="MuiBox-root MuiBox-root-6 makeStyles-tag-1 makeStyles-moderate-3"
  data-testid="complexity"
>
  Moderate
</div>
`;

exports[`Complexity Component should render Simple 1`] = `
<div
  className="MuiBox-root MuiBox-root-5 makeStyles-tag-1 makeStyles-simple-2"
  data-testid="complexity"
>
  Simple
</div>
`;
